SurveyAppTests/Tests/Services/Network/API/BaseAPITests.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  BaseAPITests.swift
+//  SurveyAppTests
+//
+//  Created by Chananchida F. on 6/17/21.
+//
+
+import Quick
+import Nimble
+import OHHTTPStubs
+
+@testable import SurveyApp
+
+enum Defines {
+
+    static let baseURL: String = "https://baseurl.com"
+}
+
+class BaseAPITests: QuickSpec {
+
+    override func spec() {
+        
+        let keychain = KeychainStorage.default
+        let userSession = UserSession(keychain: keychain)
+        var api: BaseAPI!
+        
+        let sampleSuccessData = JSON.AuthenticationService.sampleUserCredential
+        let sampleFailureData = JSON.APIError.sampleAPIError
+        
+        describe("BaseAPI") {
+            
+            context("when making a valid request") {
+                it("returns a decoded object") {
+                    api = BaseAPI(userSession: userSession)
+
+                    stub(condition: isMethodPOST()) { _ in
+                        HTTPStubsResponse(data: sampleSuccessData, statusCode: 200, headers: [:])
+                    }
+
+                    let configuration = RequestConfiguration(
+                        method: .post,
+                        url: "\(Defines.baseURL)/test"
+                    )
```

Comment:
I think we should put it in `beforeEach`, so that we can spam `it` statement if there are more one thing need to check

Reply:
And same with another `context` 😁

Reply:
Thanks a lot. I'll update the code 🙏